mycli/main.py

```diff
@@ -643,18 +640,26 @@ def one_iteration(document=None):
             if not self.less_chatty:
                 self.output('Goodbye!')
 
-    def output(self, text, **kwargs):
-        special.write_tee(text)
-        if self.logfile:
-            self.logfile.write(utf8tounicode(text))
-            self.logfile.write('\n')
-        click.secho(text, **kwargs)
 
-    def output_via_pager(self, text):
+    def output(self, output, **kwargs):
+        special.write_tee(output)
         if self.logfile:
-            self.logfile.write(text)
+            self.logfile.write(utf8tounicode(output))
             self.logfile.write('\n')
-        click.echo_via_pager(text)
+
+        size = self.cli.output.get_size()
+
+        pager = False
+        for i, line in enumerate(output.splitlines()):
+            if len(line) > size.columns or i > (size.rows - 12):
```

Comment:
Can you explain the `12` a bit? We use a maximum of `8` for the completions (which includes the bottom toolbar). Then, you need to consider the prompt.

```
mycli.get_reserved_space() + mycli.prompt.get_prompt().count('\n') + 1
```